sw/device/tests/dif/dif_gpio_test.cc

```diff
@@ -0,0 +1,420 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "gtest/gtest.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/testing/mock_mmio.h"
+
+extern "C" {
+#include "gpio_regs.h"  // Generated
+#include "sw/device/lib/dif/dif_gpio.h"
+}  // extern "C"
+
+// Base class for the test fixtures in this file.
+class DifGpioTest : public testing::Test, public mock_mmio::MmioTest {};
+
+// Init tests
+class InitTest : public DifGpioTest {};
+
+TEST_F(InitTest, NullArgs) {
+  dif_gpio_config_t config{.base_addr = dev().region()};
+  dif_gpio_t out_arg_gpio;
+
+  EXPECT_EQ(dif_gpio_init(nullptr, &out_arg_gpio),
+            kDifGpioResultInvalidArgument);
+  EXPECT_EQ(dif_gpio_init(&config, nullptr), kDifGpioResultInvalidArgument);
+  EXPECT_EQ(dif_gpio_init(nullptr, nullptr), kDifGpioResultInvalidArgument);
+}
+
+TEST_F(InitTest, Init) {
+  EXPECT_WRITE32(GPIO_INTR_ENABLE_REG_OFFSET, 0);
+  EXPECT_WRITE32(GPIO_DIRECT_OE_REG_OFFSET, 0);
```

Comment:
Done. Just a clarification: this does not really clear `MASKED_OUT_*` and `MASKED_OE_*` registers per se because they are basically connected to `DIRECT_OUT` and `DIRECT_OE`.